### PR TITLE
New environment variable data format and table

### PR DESCRIPTION
**Pull request by plaindocs**
Move environment variables to YAML for more consistency.

Change display from 

image

to 

image

Before merging 

- [x] some design love
- [x] add examples
- [ ] figure out the "modifiable issue"
- [x] fix the * anchor rspec thing

### CODE REVIEW

**Comment by plaindocs** [comment]
Example content:

```
null
"83d544ccc223c157d2bf80d3f2a32982c32c3c0db8e3674820da5064783fb091"
null
null
"10"
null
"experiment1,experiment2"
"localhost:8080"
"1a222222-e999-3636-8ddd-802222222222"
"\"BUILDKITE_AGENT_META_DATA_TAGNAME=tagvalue\", \"BUILDKITE_AGENT_META_DATA_QUEUE=some-queue\""
"elastic-builders-088264dc4f9"
"6"
"tmp/capybara/**/*;coverage/**/*"
"s3://name-of-your-s3-bucket/$BUILDKITE_PIPELINE_ID/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
"/usr/local/bin"
"master"
"/var/lib/buildkite-agent/builds/agent-1/pipeline-2"
"Carol Danvers"
"cdanvers@kree-net.com"
"Carol Danvers"
"cdanvers@kree-net.com"
"everyone:platform"
"4735ba57-80d0-46e2-8fa0-b28223a86586"
"1514"
"/var/lib/buildkite-agent/builds/"
"https://buildkite.com/acme-inc/my-project/builds/1514"
null
null
null
"script/buildkite/specs"
null
"-1"
"83a20ec058e2fb00e7fa4558c4c6e81e2dcf253d"
null
"/tmp/job-env-36711a2a-711a-484e-b180-e1b3711a80cf51b18711a"
"-ffxdq"
"-v"
null
"87972451"
"production"
"deploy"
"production"
"/etc/buildkite-agent/hooks/"
"BUILDKITE_GIT_CLEAN_FLAGS"
"e44f9784-e20e-4b93-a21d-f41fd5869db9"
":hammer: Specs"
"-1"
null
"Added a great new feature"
"acme-inc"
"0"
"5"
"main"
"my_project"
"github"
"my-project"
"[{\"github.com/buildkite-plugins/docker-buildkite-plugin#v3.7.0\":{\"image\":\"node:lts-alpine3.14\"}}]"
null
"/etc/buildkite-agent/plugins/"
null
"'123' for pull request #123, or 'false' if not a pull request"
"'main', or '' if not a pull request"
"'git://github.com/acme-inc/my-project.git', or '' if not a pull request"
"'4735ba57-80d0-46e2-8fa0-b28223a86586', or '' if not a rebuild"
"'1514', or '' if not a rebuild"
"+refs/weird/123abc:refs/local/weird/456"
"git@github.com:acme-inc/my-project.git"
"/tmp/buildkite-git-mirrors"
"0"
"AKIAIOSFODNN7EXAMPLE"
"https://buildkite-artifacts.example.com/"
null
null
"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
null
"/bin/bash -e -c"
null
null
"080b7d73-986d-4a39-a510-b34f9faf4710"
"tests-06"
"v1.2.3"
"'15' for 15 minutes, or 'false' if no timeout is set"
"datadog"
"'5aa7c894-c8c0-435b-bc17-13923b90f163', or ''' if the build was not triggered from another build"
"'1264', or '' if the build was not triggered from another build"
"'build-and-test', or '' if the build was not triggered from another build"
"Carol Danvers"
"carol@nasa.gov"
"4735ba57-80d0-46e2-8fa0-b28223a86586"
"everyone:platform"
null
```

**Comment by lachlanhardy** [comment]
image

New design in. And test fixed!

**Comment by plaindocs** [comment]
I'll have to integrate https://github.com/buildkite/docs/pull/1403 before we merge this.

**Comment by lachlanhardy** [comment]
Added `example` and styled them

image

**Comment by lachlanhardy** [comment]
@plaindocs Lemme know if I can help with that integration. Otherwise, this one is all yours.

**Comment by plaindocs** [comment]
Yep, conflict and update totes on my plate. But my plate, it overfloweth.

File: data/environment_variables.yaml [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+    The opposite of the value of the `no-plugins` [agent configuration option](/docs/agent/v3/configuration).
+  modifable: false
+  values:
+    - true
+    - false
+- name: BUILDKITE_PLUGINS_PATH
+  desc: |
+    The value of the `plugins-path` [agent configuration option](/docs/agent/v3/configuration).
+  modifable: false
+  example: "/etc/buildkite-agent/plugins/"
```
Comment (plaindocs) [inline data/environment_variables.yaml:303]:
```suggestion
  example: "`"/etc/buildkite-agent/plugins/"`"
```

File: data/environment_variables.yaml [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+- name: BUILDKITE_PULL_REQUEST_REPO
+  desc: |
+    The repository URL of the pull request.
+  modifiable: false
+  example: "'git://github.com/acme-inc/my-project.git', or '' if not a pull request"
+- name: BUILDKITE_REBUILT_FROM_BUILD_ID
+  desc: |
+    The UUID of the original build this was rebuilt from.
+  modifiable: false
+  example: "'4735ba57-80d0-46e2-8fa0-b28223a86586', or '' if not a rebuild"
```
Comment (plaindocs) [inline data/environment_variables.yaml:327]:
```suggestion
  example: "`"4735ba57-80d0-46e2-8fa0-b28223a86586"`, or `""` if not a rebuild"
```

File: pages/pipelines/environment_variables.md.erb [hunk 3]
```diff
@@ -10,6 +10,55 @@ For best practices and recommendations about using secrets in your environment v
...
+          <strong>Example: </strong>
+          <code><%= env['example'] -%></code>
+        </p>
+      <% end -%>
+      </td>
+    </tr>
+  <% end -%>
+  </tbody>
+</table>
+=======
```
Comment (karensawrey) [inline pages/pipelines/environment_variables.md.erb:61]:
```suggestion

```

**Review by karensawrey — CHANGES_REQUESTED** [review]
L53 bothers me with that "TODO" + there was 2 merge commits that I resolved & I need a 2nd pair of eyes to take a look at the results. @plaindocs

**Review by plaindocs — COMMENTED** [review]
This has been quite the journey. Let's merge it quick before anything else changes.

**Review by karensawrey — APPROVED** [review]

**Pull request merged**